src/librustdoc/clean/mod.rs

```diff
@@ -222,7 +222,7 @@ impl Clean<Lifetime> for hir::Lifetime {
             | rl::Region::Free(_, node_id),
         ) = def
         {
-            if let Some(lt) = cx.lt_substs.get(&node_id).cloned() {
+            if let Some(lt) = cx.substs.get(&node_id).and_then(|p| p.as_lt()).cloned() {
```

Comment:
I could've made this `unwrap` the Option returned by `as_lt` since it's a rustdoc bug for it to be None here, but it seemed simpler to just use `and_then`. Let me know if you want me to use `unwrap` as a sort of assertion. (Likewise with `as_ty` below.)

Reply:
I guess it's fine... Maybe add a FIXME on it so we can come back to it later on?

Reply:
Let's either leave it as-is or switch it to `unwrap`. I don't think it's worth a FIXME either way, and it's not actionable.